WordPress/Classes/Models/BasePost.m

```diff
@@ -181,7 +182,7 @@ - (void)setDateCreated:(NSDate *)localDate
      If the date is now or in the past, and the status is scheduled, set the status
      to published.
      */
-    if (self.date_created_gmt == nil) {
+    if (self.date_created_gmt == nil || [self.date_created_gmt isEqualToDate:self.dateModified]) {
```

Comment:
Also kind of related to this discussion, and maybe superseding it...

> BasePost should only contain properties and methods in common with AbstractPost and ReaderPost.

Maybe we should consider adding `dateModified` to `AbstractPost` in this PR, considering it's not used in `ReaderPost`?